common/src/main/java/com/microsoft/identity/common/AndroidPlatformComponents.java

```diff
@@ -147,4 +208,28 @@ public synchronized IDevicePopManager getDefaultDevicePopManager() throws Client
                 exception
         );
     }
+
+    @SuppressWarnings(WarningType.rawtype_warning)
+    @Override
+    public @NonNull IAuthorizationStrategyFactory getAuthorizationStrategyFactory() {
+        return AndroidAuthorizationStrategyFactory.builder()
+                .context(mContext)
+                .activity(mActivity)
+                .fragment(mFragment)
+                .build();
+    }
+
+    @Override
+    public @NonNull IStateGenerator getStateGenerator() {
+        if (mActivity == null) {
+            throw new IllegalStateException("StateGenerator requires an activity");
```

Comment:
It looks like we're creating, functionally, two separate objects here that are differentiated in behavior by their initialization parameters.  Do we want to throw this exception here, or have our static factory method return a different implementation that might not have an interface that exposes this getStateGenerator method so that we can avoid accidentally hitting this exception?

Reply:
In Android, this seems to be the case. This does/might not be true in Linux though.

IMO separating interface for this might do more harm than good - as in, we'll have to do extra casting work, and at the point we need to use this, and we try to cast the object and it failed, then we'll have to throw an exception anyway.